packages/eas-cli/src/commands/device/rename.ts

```diff
@@ -0,0 +1,162 @@
+    let appleTeamName;
+
+    if (paginatedQueryOptions.json) {
+      enableJsonOutput();
+    }
+
+    let appleTeam: AppleTeamFragment;
+
+    if (!appleTeamIdentifier) {
+      appleTeam = await selectAppleTeamOnAccountAsync(graphqlClient, {
+        accountName: account.name,
+        selectionPromptTitle: `What Apple team would you like to list devices for?`,
+        paginatedQueryOptions,
+      });
+      appleTeamIdentifier = appleTeam.appleTeamIdentifier;
+      appleTeamName = appleTeam.appleTeamName;
+    }
+
+    assert(appleTeamIdentifier, 'No team identifier is specified');
+
+    const chosenDevice = udid
+      ? await AppleDeviceQuery.getByDeviceIdentifierAsync(graphqlClient, account.name, udid)
+      : await selectAppleDeviceOnAppleTeamAsync(graphqlClient, {
+          accountName: account.name,
+          appleTeamIdentifier,
+          selectionPromptTitle: `Which device would you like to rename?`,
+          paginatedQueryOptions,
+        });
+
+    const newDeviceName = name ? name : await this.promptForNewDeviceNameAsync(chosenDevice.name);
+
+    this.logChosenDevice(chosenDevice, appleTeamName, appleTeamIdentifier, paginatedQueryOptions);
+
+    await this.renameDeviceOnExpoAsync(graphqlClient, chosenDevice, newDeviceName!);
+
+    await this.renameDeviceOnAppleAsync(chosenDevice, appleTeamIdentifier, newDeviceName!);
+  }
+
+  async promptForNewDeviceNameAsync(
+    initial: Maybe<string> | undefined
+  ): Promise<string | undefined> {
+    const { name } = await promptAsync({
+      type: 'text',
+      name: 'name',
+      message: 'New Device name:',
+      initial: initial ? initial! : undefined,
+    });
+    return name;
+  }
+
+  async renameDeviceOnExpoAsync(
+    graphqlClient: ExpoGraphqlClient,
+    chosenDevice: AppleDevice | AppleDeviceQueryResult,
+    newDeviceName: string
+  ): Promise<void> {
+    const removalSpinner = ora(`Renaming Apple device on Expo`).start();
+    try {
+      await AppleDeviceMutation.updateAppleDeviceAsync(graphqlClient, chosenDevice.id, {
+        name: newDeviceName,
+      });
+      removalSpinner.succeed('Renamed Apple device on Expo');
+    } catch (err) {
+      removalSpinner.fail();
+      throw err;
+    }
+  }
+
+  async renameDeviceOnAppleAsync(
+    device: AppleDevice | AppleDeviceQueryResult,
+    appleTeamIdentifier: string,
+    newDeviceName: string
+  ): Promise<void> {
+    const ctx = await authenticateAsync({ teamId: appleTeamIdentifier });
+    const context = getRequestContext(ctx);
+
+    Log.addNewLineIfNone();
+    const removeAppleSpinner = ora('Renaming device on Apple').start();
+    try {
+      const appleValidatedDevices = await Device.getAsync(context);
+      const appleValidatedDevice = appleValidatedDevices.find(d => d.id === device.id);
+      if (appleValidatedDevice) {
+        await appleValidatedDevice.updateAsync({ name: newDeviceName });
+      }
+      removeAppleSpinner.succeed('Renamed device on Apple');
```

Comment:
Shouldn't the message be different than "Renamed device on Apple" if we are not able to find the device on the list?

Reply:
Good call! This also exposed that something was off with the device search, and it wasn't actually renaming 🙈 . If you have a chance to peek at the new verbiage here, much appreciated! I'm going to check device:delete, as well, because it uses the same logic.